src/lu_decomposition.c

```diff
@@ -0,0 +1,48 @@
+#include "lu_decomposition.h"
+#include "matrix.h"
+#include <stdio.h>
+#include <string.h>
+
+/**
+ * @brief Perform the LU decomposition algorithm on the matrix initialMatrix
+ *
+ * @param initialMatrix The matrix to decompose
+ * @param lMatrix Output matrix L
+ * @param uMatrix Output matrix U
+ * @param size Size of the matrix
+ */
+void LUDecomposition(const lu_real* initialMatrix, lu_real* lMatrix,
+                     lu_real* uMatrix, const int size) {
+  lu_real tmpMatrix[size * size];
+  memcpy(tmpMatrix, initialMatrix, size * size * sizeof(lu_real));
```

Comment:
Could you create a variable `sizeSquared = size * size` to reduce multiplications?